### DISCUSSION TITLE
Feature Request: Ability to search for crates by author on crates.io

### DISCUSSION THREAD

**Discussion by ghost** [rust-lang/crates.io#9989] [Feature Requests]
### Summary
I'd like to request a feature that allows users to search for crates on crates.io by author (e.g., GitHub username or crates.io username).

### Motivation
Currently, crates.io allows searching by crate name, keywords, and other metadata, but there is no way to search for all crates published by a specific author. This functionality would be very useful for:

- Developers who want to explore all the crates published by a particular author they trust.
- Creators who want to showcase their work on crates.io.
- Organizations who publish multiple crates under a single account.

### Proposed Solution
A new API endpoint, something like:

`GET https://crates.io/api/v1/users/{username}/crates`

This endpoint would return a list of all crates published by the specified user. The feature could also be integrated into the search functionality on the crates.io website.

### Additional Context
I've been working on a tool that fetches crates by author, but I've found that the current public API does not support this use case. Implementing this feature natively on crates.io would benefit many in the Rust community.

Thank you for considering this request!

Best regards,  
Ben Santora (bensatlantik)  
<bensatlantik@gmail.com>

**Comment by Turbo87** [comment]
you mean something like https://crates.io/api/v1/crates?user_id=3618?

**Reply by ghost** [reply]
Actually, I just discovered that while searching for my crates with my GitHub name "bensatlantik" doesn't return results, using my crates.io name "Ben Santora" does work. That's good enough for me, but thanks for link.
